### DISCUSSION TITLE
Allow installation from bottle hosted on Github Releases

### DISCUSSION THREAD

**Discussion by yesseruser** [Homebrew/discussions#5604] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
brew config
HOMEBREW_VERSION: 4.3.21-18-g1f306b4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1f306b4d150cf1c72791786d6180a6ab83aea03d
Last commit: 23 hours ago
Core tap HEAD: 7a1b737aab6414df34d214eb75167c4cd68c6c97
Core tap last commit: 17 minutes ago
Core tap JSON: 15 Sep 00:43 UTC
Core cask tap HEAD: 0da9dd235bb9e38e65e0a86093a13cf92b5197de
Core cask tap last commit: 2 hours ago
Core cask tap JSON: 15 Sep 00:43 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_EDITOR: /opt/homebrew/bin/nvim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are not readable:
  yesser-studios/yesserstudios/yesser-todo-cli: /opt/homebrew/Library/Taps/yesser-studios/homebrew-yesserstudios/Formula/yesser-todo-cli.rb:9: unknown regexp options - gthb
    root_url https://github.com/yesser-studios/yesser-to...
                    ^~~~~~~
/opt/homebrew/Library/Taps/yesser-studios/homebrew-yesserstudios/Formula/yesser-todo-cli.rb:9: unexpected fraction part after numeric literal
...odo-cli/releases/download/1.0.2
...                             ^~
```

### Description of issue

Hi, I'm trying to let users download my formula from a bottle. I am pushing the formula to my own tap.

I have already built the bottle and hosted it on my project's Github Releases: https://github.com/yesser-studios/yesser-todo-cli/releases/download/1.0.2/yesser-todo-cli--1.0.2.arm64_sonoma.bottle.1.tar.gz

I am trying to link this bottle to my formula but I'm not sure on how to do that.

**Comment by yesseruser** [comment, marked as answer]
I figured it out.

The `brew bottle` command generates a snippet starting with `bottle do`. I copied this into the formula and added `root_url https://github.com/yesser-studios/yesser-todo-cli/releases/download/1.0.2` to the top of the snippet under `bottle do`.

I had to rename the bottle, because the name contained two dashes before 1.0.2.

**Comment by carlocab** [comment]
This post might be useful for you:

https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

TL;DR: Use the workflows created by `brew tap-new`. That should handle bottle creation (on GitHub Actions Runners) and uploading to GitHub Releases.
